core/shared/src/main/scala/zio/ZLayer.scala

```diff
@@ -351,6 +357,12 @@ sealed abstract class ZLayer[-RIn, +E, +ROut] { self =>
   ): ZLayer[RIn, E1, ROut2] =
     self >>> that
 
+  @deprecated("For macros only. Use to")
+  final def toEager[E1 >: E, ROut2](that: ZLayer[ROut, E1, ROut2])(implicit
+    trace: Trace = Trace.empty
```

Comment:
We could probably generate at the top of the generated code and then it should be fine.